securedrop/debian/app-code/usr/bin/securedrop-set-redis-auth.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
+
+import argparse
+import re
+import secrets
+import shutil
+import sys
+from pathlib import Path
+from typing import Optional
+
+CONFIG_PY = Path("/var/www/securedrop/config.py")
+RQ_CONFIG_PY = Path("/var/www/securedrop/rq_config.py")
+REDIS_CONF = Path("/etc/redis/redis.conf")
+
+PYTHON_RE = re.compile(r"^REDIS_PASSWORD = ['\"](.*?)['\"]$")
+REDIS_CONF_RE = re.compile(r"^requirepass (.*?)$")
+
+
+def read_python_file(path: Path) -> Optional[str]:
+    """Extract the password from a Python file"""
+    if not path.exists():
+        # rq_config.py might not exist yet
+        return None
+    contents = path.read_text()
+    # Read in reverse because we want to look for the last matching line
+    # since it'll take precedence in Python
+    for line in contents.splitlines()[::-1]:
+        match = PYTHON_RE.match(line)
+        if match:
+            return match.group(1)
+    # Nothing found
+    return None
+
+
+def write_python_file(path: Path, password: str) -> None:
+    """Set the new password in a Python file, removing any existing passwords"""
+    if path.exists():
+        lines = path.read_text().splitlines()
+    else:
+        lines = []
+    # Take the existing file, remove any matching password lines, and then add
+    # our new password at the end
+    lines = [line for line in lines if not PYTHON_RE.match(line)]
+    lines.append(f"REDIS_PASSWORD = '{password}'")
+    if path == RQ_CONFIG_PY and not path.exists():
+        # Ensure rq_config.py is created with the correct permissions
+        path.write_text("")
+        path.chmod(0o640)
+        shutil.chown(path, "root", "www-data")
+
+    path.write_text("\n".join(lines) + "\n")
+
+
+def read_redis_conf(path: Path) -> Optional[str]:
+    """Extract the password from our redis.conf file"""
```

Comment:
Shouldn't this also have (from `read_python_file()`):

```suggestion
    """Extract the password from our redis.conf file"""
    if not path.exists():
        # redis.conf might not exist yet
        return None
```

Reply:
No, we should be able to safely assume that redis.conf was installed by the package, so failing loudly here if it doesn't exist is the right way to go. I can leave a comment to that effect though.

Reply:
Probably writing a blank redis.conf with just a password would be a bad thing, and this is implicitly making sure that doesn't happen, so let me add an explicit check that it *does* exist.

Reply:
Done.